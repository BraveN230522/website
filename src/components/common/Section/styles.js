import styled from '@emotion/styled';

const getGridStyles = ({ grid, columns, theme }) =>
  grid &&
  `
  display: grid;
  grid-gap: 17px;
  grid-template-columns: repeat(${columns}, 1fr);

  ${theme.media.tablet`
    grid-template-columns: repeat(3, 1fr);
  `}

  ${theme.media.phone`
    grid-template-columns: repeat(2, 1fr);
  `}
`;

const SectionStyles = styled.div`
  background-color: ${({ dark, theme }) => (dark ? `${theme.colors.black}` : `${theme.colors.white}`)};
  ${({ centered, theme }) => (centered ? theme.setContainer() : 'width:100%;')}
`;

const TitleHeader = styled.header`
  display: flex;
  justify-content: space-between;
  margin: 0 0 16px;
`;

const Title = styled.h1`
  color: ${({ dark, theme }) => (dark ? `${theme.colors.white}` : `${theme.colors.black}`)};
  font-family: ${({ theme }) => theme.fonts.Inter};
  font-size: 22px;
  font-weight: 400;
  letter-spacing: -0.15px;
  line-height: 29px;
  margin: 0;

  ${({ theme }) =>
    theme.generateBreakpoint(
      378,
      `
        font-size: 20px;
      `
    )};
`;

const Article = styled.article`
  align-items: center;
  display: flex;

  ${getGridStyles}

  ${({ theme }) => theme.media.phone`
      grid-gap: 8px;
  `}
`;

export { SectionStyles, TitleHeader, Title, Article };
