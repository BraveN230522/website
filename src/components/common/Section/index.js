import React from 'react';

import { SectionStyles, TitleHeader, Title, Article } from './styles';

const Section = ({ title, more, children, columns, grid, className, centered, dark }) => {
  return (
    <SectionStyles className={className} centered={centered} dark={dark}>
      <TitleHeader>
        <Title dark={dark}>{title}</Title>
        {more}
      </TitleHeader>

      <Article columns={columns} grid={grid}>
        {children}
      </Article>
    </SectionStyles>
  );
};

export default Section;
