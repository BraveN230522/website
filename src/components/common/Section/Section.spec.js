import React from 'react';
import { render } from '@testing-library/react';
import withGlobals from '@swiamTestUtils/withGlobals';
import Section from './index';

describe('<Section />', () => {
  it('Renders section Component', () => {
    const { container } = render(<Section />, {
      wrapper: withGlobals,
    });
    expect(container).toMatchSnapshot();
  });
});
