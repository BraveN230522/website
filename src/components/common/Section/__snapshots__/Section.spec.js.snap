// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> Renders section Component 1`] = `
.emotion-6 {
  background-color: #ffffff;
  width: 100%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0 0 16px;
}

.emotion-0 {
  color: #000000;
  font-family: Inter,sans-serif;
  font-size: 22px;
  font-weight: 400;
  -webkit-letter-spacing: -0.15px;
  -moz-letter-spacing: -0.15px;
  -ms-letter-spacing: -0.15px;
  letter-spacing: -0.15px;
  line-height: 29px;
  margin: 0;
}

@media (max-width:23.625em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
  <div
    class="emotion-6 emotion-7"
  >
    <header
      class="emotion-2 emotion-3"
    >
      <h1
        class="emotion-0 emotion-1"
      />
    </header>
    <article
      class="emotion-4 emotion-5"
    />
  </div>
</div>
`;
